kfs-core/src/main/java/edu/arizona/kfs/fp/service/impl/ProcurementCardServiceImpl.java

```diff
@@ -0,0 +1,205 @@
+            }
+        }
+        return success;
+    }
+    
+    private Date getProcurementCardTransactionPostingDetailDate(ProcurementCardDocument procurementCardDocument) {
+        Date date = null;
+        
+        if(!procurementCardDocument.getTransactionEntries().isEmpty()) {
+            date = ((ProcurementCardTransactionDetail)procurementCardDocument.getTransactionEntries().get(0)).getTransactionPostingDate();
+        }
+        
+        return date;
+    }
+    
+    /**
+     * For the given PCDO and specific accounting line create and add pending entries for use tax based on the
+     * tax region rate given.
+     */
+    protected boolean generateUseTaxPendingEntries(ProcurementCardDocument procurementCardDocument, ProcurementCardTargetAccountingLine sourceDetail, GeneralLedgerPendingEntrySequenceHelper sequenceHelper, 
+            TaxRegion taxRegion, TaxRegionRate taxRegionRate) {
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("generateUseTaxEntries started");
+        }
+        //Is the Object Code Non Taxable?
+        String financialObjectCode = sourceDetail.getFinancialObjectCode();
+        if (StringUtils.isBlank(financialObjectCode) || parameterEvaluatorService.getParameterEvaluator(ProcurementCardDocument.class, NON_TAXABLE_OBJECT_CODES, financialObjectCode).evaluationSucceeds()) {
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("isUseTaxTransaction() Use Tax Excluded Object Code - not posted");
+            }
+            return true;
+        }
+        
+        KualiDecimal generatedTransactionAmount = new KualiDecimal(sourceDetail.getAmount().bigDecimalValue().multiply(taxRegionRate.getTaxRate())).abs();
+        
+        GeneralLedgerPendingEntry e = new GeneralLedgerPendingEntry();
+        generalLedgerPendingEntryService.populateExplicitGeneralLedgerPendingEntry(procurementCardDocument, sourceDetail, sequenceHelper, e);
+        e.setTransactionLedgerEntryAmount(generatedTransactionAmount);
+        e.setTransactionLedgerEntryDescription(getUseTaxDescription(sourceDetail.getAmount(), taxRegionRate.getTaxRate()));
+        procurementCardDocument.customizeExplicitGeneralLedgerPendingEntry(sourceDetail, e);
+        procurementCardDocument.addPendingEntry(e);
+        sequenceHelper.increment();
+    
+        // handle the offset entry
+        GeneralLedgerPendingEntry offsetEntry = new GeneralLedgerPendingEntry(e);
+        generalLedgerPendingEntryService.populateOffsetGeneralLedgerPendingEntry(procurementCardDocument.getPostingYear(), e, sequenceHelper, offsetEntry);
+        procurementCardDocument.customizeOffsetGeneralLedgerPendingEntry(sourceDetail, e, offsetEntry);
+    
+        procurementCardDocument.addPendingEntry(offsetEntry);
+        sequenceHelper.increment();
+            
+        //Create the entry and offset for the Use Tax account
+        e = new GeneralLedgerPendingEntry(e);
```

Comment:
Use a different var than e that has a suggestive name referring on what it will be used for.